src/RedisDiscover.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#include <string.h>
+#include <vector>
+#include <algorithm> 
+#include <chrono>
+#include <thread>
+extern "C" {
+    #include <hiredis/hiredis.h>
+}
+
+class RedisDiscover {
+    redisContext *redis_client_;
+    std::string tasks_list_;
+    
+    std::string getRedisHostNameFromEnv();
+
+    public:
+        RedisDiscover(): redis_client_(NULL), tasks_list_("deku_tasks") {};
+        ~RedisDiscover();
```

Comment:
this is the kind of wrapper i was referring to in my other comment.

this class would need a copy, or at least a move constructor. if you do not implement those, delete them:
RedisDiscover(RedisDiscover const&) = delete;
so it does not get accidentally called.